packages/node_modules/@node-red/editor-client/src/js/ui/editor.js

```diff
@@ -148,12 +148,23 @@ RED.editor = (function() {
             ((typeof definition[property].label) == "string")) {
             label = definition[property].label;
         }
-        if ("required" in definition[property] && definition[property].required) {
-            valid = value !== "";
-            if (!valid && label) {
-                return RED._("validator.errors.missing-required-prop", {
-                    prop: label
-                });
+        if ("required" in definition[property]) {
+            if (definition[property].required) {
+                valid = value !== "";
+                if (!valid && label) {
+                    return RED._("validator.errors.missing-required-prop", {
+                        prop: label
+                    });
+                }
+            } else {
+                if (value === "") {
+                    return true;
+                }
+            }
+            
+        } else {
+            if (value === "") {
+                return true;
             }
```

Comment:
I'm hesitant because it can be validated in a custom function - I don't know what an undefined value should do.

In my nodes I don't define it (for some properties) because it's the custom validator which handle it.